app/models/game.rb

```diff
@@ -49,9 +49,42 @@ def render_piece(x, y)
 
   def check?(is_white)
     king = pieces.find_by(type: KING, is_white: is_white)
+    return false if king.nil?
     pieces.where(game_id: id, is_white: !is_white).find_each do |piece|
+      next if piece.x_position == nil || piece.y_position == nil
       return true if piece.valid_move?(king.x_position, king.y_position)
     end
     false
   end
+
+  def stalemate?(is_white)
+    return false if check?(is_white)
+    (1..8).each do |new_x|
+      (1..8).each do |new_y|
+        pieces.where(is_white: is_white).each do |piece|         
+          begin
+            piece.move_to!(new_x, new_y)
+          #   next unless piece.actual_move?(new_x, new_y)
+          #   next unless piece.valid_move?(new_x, new_y)
+          #   if square_occupied?(new_x, new_y)
+          #     occupying_piece = Piece.get_piece_at_coor(new_x, new_y)
+          #     if (occupying_piece.is_white && is_white?) || (!occupying_piece.is_white && !is_white?)
+          #       next
+          #     else
+          #       capture_piece(occupying_piece)
+          #     end
+          #   elsif piece.type == PAWN 
+          rescue ArgumentError
+            next
+          else
+            binding.pry
+            return false
+            raise ActiveRecord::Rollback
+          end
+        end
```

Comment:
Take a look at `.transaction`. Could wrap the block in a transaction, then roll it back if it fails.

http://markdaggett.com/blog/2011/12/01/transactions-in-rails/

Note: This might cause an issue in tests, because each test runs in a transaction (might work fine though, I can't remember)